hack/dev-setup-register-gardener

```diff
@@ -19,15 +19,35 @@ source $(dirname "${0}")/common
 IP_ROUTE=$(ip route get 1)
 IP_ADDRESS=$(echo ${IP_ROUTE#*src} | awk '{print $1}')
 
+APISERVER_SERVICE_NAME="gardener-apiserver"
+APISERVER_ENDPOINT_NAME="gardener-apiserver"
+APISERVER_SERVICE_PORT=443
+APISERVICE_PORT_STRING=""
+
+APISERVER_EXTERNALNAME=kubernetes.localhost
+
 CORE_V1ALPHA1_APISERVICE_NAME="v1alpha1.core.gardener.cloud"
 CORE_V1BETA1_APISERVICE_NAME="v1beta1.core.gardener.cloud"
 SETTINGS_APISERVICE_NAME="v1alpha1.settings.gardener.cloud"
 
-APISERVER_SERVICE_NAME="gardener-apiserver"
-APISERVER_ENDPOINT_NAME="gardener-apiserver"
 CONTROLLER_MANAGER_SERVICE_NAME="gardener-controller-manager"
 CONTROLLER_MANAGER_ENDPOINT_NAME="gardener-controller-manager"
 
+if [[ $(k8s_env) == "$NODELESS" ]]; then
+  APISERVER_SERVICE_PORT=8443
+  APISERVICE_PORT_STRING="    port: $APISERVER_SERVICE_PORT"
+
+  echo "Adding API_SERVER address mapping to /etc/hosts!"
+
+  # remove entry from /etc/hosts
+  sudo sed -i".bak" "/$APISERVER_EXTERNALNAME/d" /etc/hosts
+
+  # then add it to make sure that addition is idempotent
+  LINE="$IP_ADDRESS\t$APISERVER_EXTERNALNAME"
+  sudo -- sh -c -e "echo '$LINE' >> /etc/hosts";
```

Comment:
Instead of adjusting the `hosts` file of the host, can we rather adjust only the one inside the `kube-apiserver` container?